### ISSUE TITLE
[question] Do version ranges in conanfile.txt on install work on remotes if not found locally?

### ISSUE THREAD

**Issue by dyle71** [conan-io/conan#8712]
#!/bin/hi *

My setup:
```bash
$ conan -v
Conan version 1.34.1
$ uname -a
Linux novigrad 5.8.0-48-generic #54-Ubuntu SMP Fri Mar 19 14:25:20 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.10
Release:        20.10
Codename:       groovy
```

I do have a `conanfile.txt` in my local project like this:
```
[requires]
foo/[>=1.1.0]@blah/blob
```
Which resolves to a `foo/1.1.0` on a specific remote.

Now running conan like this:
```bash
$ conan install .
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=10
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

ERROR: Version range '>=1.1.0' from requirement 'foo/[>=1.1.0]@blah/blob' required by 'conanfile.txt' could not be resolved in local cache
```

Which is fine. Though  https://docs.conan.io/en/latest/versioning/version_ranges.html#version-ranges states 

> "If no matching version is locally found, it starts to search in the remotes, in order.  "

Tweaking the requirement to 
```
[requires]
foo/1.1.0@blah/blob
```
will now download from the remote.

And a successive call to `conan install .` using the version range as before, now succeeds with:
```
Version ranges solved
    Version range '>=1.1.0' required by 'conanfile.txt' resolved to 'foo/[>=1.1.0]@blah/blob' in local cache
```

In summary: version ranges do NOT work on remotes but only on local cache. Is this true, intentional?

/me == confused.

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
I am not sure what is failing in your case. I have done this test, and it seems to work here:

conanfile.txt


```bash
C:\Users\memsharded\example
(conan36) λ conan remove * -f                                                                             
                                                                                                          
C:\Users\memsharded\example                                                                          
(conan36) λ conan search                                                                                  
There are no packages                                                                                     
                                                                                                          
C:\Users\memsharded\example                                                                            
(conan36) λ cat conanfile.txt                                                                             
[requires]                                                                                                
zlib/[>=1.0]                                                                                              
                                                                                                          
C:\Users\memsharded\example                                                                             
(conan36) λ conan install conanfile.txt                                                                   
Configuration:                                                                                            
[settings]                                                                                                
arch=x86_64                                                                                               
arch_build=x86_64                                                                                         
build_type=Release                                                                                        
compiler=Visual Studio                                                                                    
compiler.runtime=MD                                                                                       
compiler.version=16                                                                                       
os=Windows                                                                                                
os_build=Windows                                                                                          
[options]                                                                                                 
[build_requires]                                                                                          
[env]                                                                                                     
                                                                                                          
zlib/1.2.11: Not found in local cache, looking in remotes...                                              
zlib/1.2.11: Trying with 'conan-center'...                                                                
Decompressing conan_export.tgz completed [0.00k]                                                          
zlib/1.2.11: Downloaded recipe revision b291478a29f383b998e1633bee1c0536                                  
Version ranges solved                                                                                     
    zlib/* versions found in 'conan-center' remote                                                        
    Version range '>=1.0' required by 'conanfile.txt' resolved to 'zlib/1.2.11' in remote 'conan-center'  
                                                                                                          
conanfile.txt: Installing package                                                                         
Requirements                                                                                              
    zlib/1.2.11 from 'conan-center' - Downloaded                                                          
Packages                                                                                                  
    zlib/1.2.11:3fb49604f9c2f729b85ba3115852006824e72cab - Download                                       
                                                                                                          
Installing (downloading, building) binaries...                                                            
zlib/1.2.11: Retrieving package 3fb49604f9c2f729b85ba3115852006824e72cab from remote 'conan-center'       
Decompressing conan_package.tgz completed [0.00k]                                                         
zlib/1.2.11: Package installed 3fb49604f9c2f729b85ba3115852006824e72cab                                   
zlib/1.2.11: Downloaded package revision 4f7de2236af7b705df9ca91396dc6ba4                                 
conanfile.txt: Generator txt created conanbuildinfo.txt                                                   
conanfile.txt: Generated conaninfo.txt                                                                    
conanfile.txt: Generated graphinfo                                                                                                                                                          
```

Subsequents ``conan install conanfile.txt`` also work (and faster, it founds in cache).

Could you please try the above steps? Not sure what could be failing in your case, but this should work.
Maybe you have some front like nginx in front of your server, with something that could be affecting the ``search`` endpoint of the protocol (that is used for the range-resolver for version ranges)?. Maybe some permissions in the repo? 
It might be worth launching a local ArtifactoryCE in your computer (it is easy, download the zip, unzip and run the launcher), to rule out possible server side issues that could be affecting.

**Comment by dyle71** [comment]
Thank you James.

Now, these steps work flawless on my setup too. But I wrote that already. That's not the point. I'm already some steps beyond that.

I may have pointed out more clearly the fact, that I'm downloaded from a _specific remote_ (to be concrete: it's the conan repo service at GitLab). This is not the conan-center.

In short, this is the problem:

* `package/1.1.0@foo/bar` resides on a specific remote https://some.example.repo
* I've added the remote to my local conan setup
* Stating `package/[>=1.0.0]@foo/bar` in the `conanfile.txt` now **fails**.
* But stating `package/1.1.0@foo/bar` **works** and puts the package in the local cache.
* A concluding `package/[>=1.0.0]@foo/bar` in the `conanfile.txt` again **works** since it is found in the local cache.

When I do this against the conan center, it works flawless right from the start. But not on the remote. If and only if I work with a version range in my `conanfile.txt`.

But maybe this is a problem from the remote, since I encountered recently, that this remote also serves the package metadata files but refuses to ship the package binaries themselves. See, the metadata json in the local cache is valid but the package node in there is empty `{}`. I found similar github issues about the latter years back.

It may be due that the conan API is not properly implemented at the remote site causing the query on `package/[>=1.0.0]@foo/bar` to fail.

So currently I'm striving to set up my very own repo server and move from that other remote. I have the suspicion that the former one is buggy and I want to verify this assumption.

**Comment by memsharded** [comment]
> I may have pointed out more clearly the fact, that I'm downloaded from a specific remote (to be concrete: it's the conan repo service at GitLab). This is not the conan-center.

Yes, this is the most relevant point, and it was missing in the previous comment, this is why I was trying to figure out what could be, and the above proposed steps always help to start investigating and ruling out possible causes.

In this case, I'd say yes, the most likely cause is a server side issue on the Gitlab repo. Please keep us posted about your investigation, thanks!

**Comment by AbrilRBS** [comment]
We would love to hear if you ever found what was going wrong serverside, but I'll close this issue as stale as we didn't hear back from you, but we're always happy to help if you ever have any other problems :)
